src/hooks/useDebouncedFetchSections.ts

```diff
@@ -47,17 +48,40 @@ const useDebouncedFetchSection = (
         .then((response) => {
           const newSectionsData: AutocompleteResultSections = {};
           Object.keys(response.sections).forEach((section: string) => {
```

Comment:
This function is getting complex. Can we abstract it into something like a `transformResponse`  util function or whatever name you would like?

Reply:
great call out. done.